kitchen/modules/execution-schedule/src/lib.rs

```diff
@@ -153,10 +150,11 @@ decl_module! {
             let voters_signal = <SignalBank<T>>::get(current_era, &voter);
             ensure!(voters_signal >= signal, "The voter cannot signal more than the remaining signal");
             if let Some(mut task) = <PendingTasks<T>>::get(id.clone()) {
-                task.score.checked_add(signal).ok_or("task is too popular and signal support overflowed")?;
+                task.score = task.score.checked_add(signal).ok_or("task is too popular and signal support overflowed")?;
+                <PendingTasks<T>>::insert(id.clone(), task);
```

Comment:
Yes, this is necessary for writing to storage. I assumed the mutability of task would lead to updates upon every change, but that is a very incorrect assumption. This seems like it could be a common mistake (while it is for me at least!)